File: 대표 알고리즘/정렬/bubble-sort.js
let array = [9, 8, 7, 6, 5, 4, 3, 2, 1]

const bubbleSort = (array) => {
  for (let i = 0; i < array.length - 1; i++) {
    for (let j = 0; j < array.length - 1 - i; j++) {
      if (array[j] > array[j + 1]) {
        const tempVal = array[j]
        array[j] = array[j + 1]
        array[j + 1] = tempVal
      }
    }
    console.log(array)
  }
}

console.log(`before : ${array}`)
bubbleSort(array)
console.log(`after : ${array}`)

export {}

File: 대표 알고리즘/정렬/bubble-sort.py
array = [9,8,7,6,5,4,3,2,1]

def bubbleSort(array):
  n = len(array)
  for i in range(n-1):
    for j in range(n-1-i):
      if array[j] > array[j+1]:
        array[j], array[j+1] = array[j+1], array[j]
      print(array)

print("before: ", array)
bubbleSort(array)
print("after: ", array)